### PR TITLE
add `.varnames` attribute to `Moran_BV` objects in `Moran_BV_Matrix` results

### PR DESCRIPTION
**Pull request by slumnitz**
adapt documentation for `.varnames`
set default `varnames` if not specified

### CODE REVIEW

**Comment by darribas** [comment]
This looks good to me. Just for discussion (maybe this is not really the best place but at least to test the apetite): could we check if `variables` is a `DataFrame` instead of a list and pull names from there? It'd be uneven in the sense other functions do not support this but there are parts on the library that support `pandas.Series` objects if I recall. What do people think?

File: esda/moran.py [hunk 1]
```diff
@@ -465,9 +465,8 @@ def Moran_BV_matrix(variables, w, permutations=0, varnames=None):
     permutations : int
                    number of permutations
     varnames     : list
-                   strings for variable names. If specified runtime summary is
-                   printed
-
+                   strings for variable names. Will add an
+                   attribute to `Moran_BV` objects in results
```
Comment (darribas) [inline esda/moran.py]: It'd be good here to add a brief explanation of what happens if `varnames=None`

**Comment by ljwolf** [comment]
I would dig the handling of the dataframes, but that'd add a function-level dependency on pandas if we use a typical `isinstance` check. Is there a way we can ducktype it? I think we could rely on the fact that `list(df)` gives a list of columns, anyway?

**Comment by slumnitz** [comment]
we could make it a soft dependency with a `try isinstance` check and then the import? Would that be an option?

**Comment by slumnitz** [comment]
Yes. Then users also would not need to input `varnames` and `variables` but we could deal with both from the one df?

**Comment by ljwolf** [comment]
I like that; if `variables` is a dataframe (and no `varnames` are given), do all pairs; otherwise, subset `variables` by `varnames` if `variables` is a df.

**Comment by slumnitz** [comment]
OK sounds good I can deal with that today.

**Comment by slumnitz** [comment]
@darribas, @ljwolf have a look if this fits better. Ps: I can remove the notebook or move it somewhere else. It is just for an illustration purpose.

File: esda/moran.py [hunk 2]
```diff
@@ -505,6 +508,35 @@ def Moran_BV_matrix(variables, w, permutations=0, varnames=None):
...
+        # check if pandas is installed
+        import pandas
+        if isinstance(variables, pandas.DataFrame):
+            # if yes use variables as df and convert to numpy_array
+            varnames = pandas.Index.tolist(variables.columns)
+            variables_n = []
+            for var in varnames:
+                variables_n.append(variables[str(var)].values)
+        else:
+            raise ImportError('install pandas in order to pass /n'
```
Comment (darribas) [inline esda/moran.py]: Won't this be raised if `variables` is not an instance of `DataFrame`? So, if it's an array, will it raise the error? As I understand the documentation, it shouldn't, right?

File: notebooks/moran_matrix_example.ipynb [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
...GDbtm15jct9990HwHXXXbes69aiyHby6GGe/vBfesaYP3o7BAIY4BnVDRs2iIouCFXgoYceAuDKK69c8LN+0ccv0uRLE9u8eTO2bRMMBkmn0xiG4dk29zyRSIRoNMrG7/wHWgEo7t32LIxUijVr1rB27VrieWoKCUI94074m5ubiUQipNNpLz08mUzWdMJfqjOSz+E6evQoqVSKLVu2LHteUilR5cS9d3uvK5E2JhHMQjVxa45qrQmFQt77tm1XvOboQixWaHEch9nZWU6cOMGxY8fy3j9uyunXvvY14vE4v/Irv0IkEvGe/0v1ERbr7+TWEfLf9+l0mqmpKVpaWrLmJYsRdGq5KC80BgVFIaXUh4FfB3YD71BK3Qu8AXg/8JpyXFwp9SXgamC9UmoQeC8QANBafwb4DvBi4AAQB24px3WFDBs3bkQpxejoqLcq7hrKfJSrFWW1i2xPPPxDxu52Mx41vOEvsva7RtUNF11NKrqsPAq14Pjx4/PeyxV/8ok+PT099Pb2opTKmyZ28OBBJicnOXnyJIDXWdEVkwYHB+np6WHTNz8PSqG14r+2PhPDMAgGg0xNTdHc3Fy26INCgpYglBvXhuerKRQIBGo64S/FGckXTTQzM0M8HicWi3HixAm6urrmCSH+eUmh51klRZVS6wgt51krEcxCNXFrjubaDMuymJmZKVhnpxqUKrT47/lEIkEymfQEc6111v3jpqIGg0GOHz/OWWedlWUXluMjlOrv+O2ImqsP50+5DQQCxONxpqenATxhaDGCTq0W5YXGoVik0PXARVrrGaXUOjJpWzu11k+V6+Ja61cssF8Df1yu6wnZ1KrYdTWve/T2jxF/co+3PfuHb8cqMDENh8OrRkWXlUehXigk/miteeqpp7JEn4GBAQC6u7vzpom5Exw3JdYVhbTWrFu3jgMHDnD2vf8Gc5MzpRRpR2Mqx7v34/E455133rJEnIUErVIQQUlYDK6QeerUqZrVFCo2toWckdwUt3g8Tjwe9/72DcMoKIRorRkeHi74PFuqqLKQkON35Dpfeeu8/e7YlvOslTogQq3IlzJWD/PDUoQW9553o4YtyyKRSADQ3Nyc9/6xLMuzm/7vuRwfoRR/59SP7sva7vvQv+S978PhsGcX3QL7ixV0at35WqhviolCCa31DIDW+pRSan85BSGhfljuqvhSV
```
Comment (darribas) [inline notebooks/moran_matrix_example.ipynb]: Quick questions (commenting here as I can't on the rendered version of the notebook):
- why are the x labels in the `moran_facette` skewed to the right? The y labels look centered and OK though.
- I'd suggest to move `moran_facette` to `moran_facet` as it's better aligned with the terminology used elsewhere.

**Comment by darribas** [comment]
Also, this is looking great @slumnitz! Only a couple of clarifications on my side.

**Comment by slumnitz** [comment]
Yes notebook will be moved to `splot` potentially added to the `esda_morans.ipynb` example

**Pull request merged**